bpf/bpf_host.c

```diff
@@ -37,6 +37,10 @@
  */
 #define SKIP_ICMPV6_ECHO_HANDLING
 
+#ifndef VLAN_FILTER
+# define VLAN_FILTER(ifindex, vlan_id) return CTX_ACT_DROP;
```

Comment:
You might want to consider adding a more complex default in the `bpf/node_config.h` file. That file is only used for testing, and is used in existing verifier tests. So if you define `VLAN_FILTER(...)` there with the most common configuration you would expect (could be 1-2 or maybe even 5 VLAN tags), then this would help us to catch potential verifier issues for common use cases before we release & you deploy them.

Reply:
Common configuration from my point of view is one main device and one vlan device - internal and external networks. We faced that problem with our setup on hetzner datacenter. Telling the truth, I can't imaging configuration with lot's of vlan devices for kubenetes.

Reply:
:+1: Given that VLANs have so much legacy usage, I suspect that in users own datacenters, additional use cases could come up with any number of VLAN configurations. I also note that by default this kind of configuration is configured once for the entire k8s cluster via cilium-config ConfigMap, and in the code here we configure it once for the entire node by generating the code into `node_config.h`. There's no sort of mapping between specific devices and specific VLAN tags. This means that the overall complexity is determined by the total number of VLAN tags that you expect your k8s cluster to encounter, and most likely every device on every node will have the same configuration.

If this is OK with you, then I see this as an acceptable limitation of the current proposed approach. If the community wants to drive a better solution in future we can always enhance the logic. On the other hand, if this is surprising to you and you expect to encounter issues with this as-is, then let's identify those issues and make sure this proposal is at least addressing your use case.

Reply:
Think that idea with just allow or drop all vlan tagged packets with default to drop is better for now. It will defenitely solve problems with vlan we're having. Also I think that 'vlan aware' part of community isn't too big and this approach will suite everybody with such problem.

Reply:
Also such approach will make much less changes to datapath.